parser.go

```diff
@@ -1488,27 +1486,70 @@ func (p *parser) parseSchemaPropertiesFromStructFields(pkgPath, pkgName string,
 		isSliceOrMap := strings.HasPrefix(typeAsString, "[]") || strings.HasPrefix(typeAsString, "map[]")
 		isInterface := strings.HasPrefix(typeAsString, "interface{}")
 		if isSliceOrMap || isInterface || typeAsString == "time.Time" {
-			var err error
-			fieldSchema, err = p.parseSchemaObject(pkgPath, pkgName, typeAsString, true)
-			if err != nil {
-				p.debug(err)
-				return
+			splitType := strings.Split(typeAsString, "]")
+			if len(splitType) > 1 {
+				valueType := splitType[1]
+				if !isBasicGoType(valueType) {
+					if _, ok := p.KnownIDSchema[valueType]; ok {
+						if strings.HasPrefix(splitType[0], "map") {
+							fieldSchema.Type = &objectType
+							fieldSchema.AdditionalProperties = &SchemaObject{Ref: addSchemaRefLinkPrefix(p.getTypeAsString(valueType))}
+						} else {
+							fieldSchema.Type = &arrayType
+							fieldSchema.Items = &SchemaObject{Ref: addSchemaRefLinkPrefix(p.getTypeAsString(valueType))}
+						}
+					} else {
+						fieldSchemaSchemeaObjectID, err := p.registerType(pkgPath, pkgName, typeAsString)
```

Comment:
Should this be `fieldSchemaSchemaObjectID`? It's a common typo in this repo, but we should try not to copy-paste it at least :P